### DISCUSSION TITLE
Why I'm getting an infinite rendering on simple Counter component?

### DISCUSSION THREAD

**Discussion by abhinav-anshul** [pmndrs/jotai#1561] [Q&A] [answered]
Hi,

I'm using Jotai for the very first time, hence created a simple counter component, which I believe is correct :

```
import { useAtom, atom } from "jotai"

function Counter() {
  const AtomCount = atom(10)
  const [count, setCount] = useAtom(AtomCount)
  console.log({ AtomCount })
  return (
    <>
      <button onClick={() => setCount((c) => c + 1)}></button>
      <div>{count}</div>
      <button onClick={() => setCount((c) => c - 1)}></button>
    </>
  )
}

export default Counter

```

However, this logs the value of AtomCount infinite times in the devtools, hence must be rendering as well. Why is this happening?

**Comment by dai-shi** [comment, marked as answer]
Please define atoms outside components. Otherwise, it will create new atoms indefinitely.
